YarpJS/include/YarpJS_Callback.h

```diff
@@ -130,10 +131,22 @@ void YarpJS_Callback<T>::_internal_worker_end(uv_work_t *req, int status)
 template <class T>
 void YarpJS_Callback<T>::callCallback()
 {
+    mutex_callback.lock();
 
+    if(callback!=NULL)
+        uv_async_send( &(this->async) );
```

Comment:
It is probably clear for you, but just for me for understanding: in the case `callback!=NULL`, who/where is being called `mutex_callback.unlock()` that match the earlier `mutex_callback.lock()`?

Reply:
It is called in `YarpJS_Callback<T>::_internal_worker_end(...)`, which were scheduled by `YarpJS_Callback<T>::_internal_async`, itself scheduled by `uv_async_init` in `YarpJS_Callback<T>::setCallback`.

Those are then triggered through `uv_async_send()` in case `callback!=NULL`.

Reply:
Sorry, just realized I explained the same as before in https://github.com/robotology/yarp.js/pull/33#pullrequestreview-851910443, so maybe you want additional details... let me know.

Reply:
No, thanks this is super clear, I had forgot about https://github.com/robotology/yarp.js/pull/33#pullrequestreview-851910443 .